lib/package-manager/install.js

```diff
@@ -4,25 +4,33 @@ const path = require('path');
 const stripAnsi = require('strip-ansi');
 
 const createOptions = require('../create-options');
+const getExecutable = require('./get-executable');
 const spawn = require('../spawn');
 const timeout = require('../timeout');
 
-function install(context, next) {
+function install(packageManager, context, next) {
   const options =
     createOptions(
       path.join(context.path, context.module.name), context);
   let args = ['install'];
-  context.emit('data', 'info', context.module.name + ' npm:',
-      'npm install started');
+  context.emit('data', 'info', context.module.name + ' ' + packageManager +
+    ':',
+    packageManager + ' install started');
 
-  context.emit('data', 'verbose', context.module.name + ' npm:',
+  context.emit('data', 'verbose', context.module.name + ' ' + packageManager +
+    ':',
     'Using temp directory: "' + options.env['npm_config_tmp'] + '"');
 
   if (context.module.install) {
     args = args.concat(context.module.install);
   }
 
-  const proc = spawn('npm', args, options);
+  const packageManagerBin = getExecutable(packageManager, options);
```

Comment:
afaict this now introduces something that is sync into a hot path (the install. Is there not a way to do this once during initialization rather then each install.

I'd like to propose that we change the `packageManager` argument to have the following pattern
```
{
  "name":  "[npm]|[yarn]"
  "path": "/path/to/binary"
}
```

we can cache the async call to the path in the initialization